src/basilisp/lang/map.py

```diff
@@ -107,6 +117,66 @@ def to_persistent(self) -> "PersistentMap[K, V]":
         return PersistentMap(self._inner.finish())
 
 
+def map_lrepr(
+    entries: Callable[[], Iterable[Tuple[Any, Any]]],
+    start: str,
+    end: str,
+    meta: Optional[IPersistentMap] = None,
+    **kwargs: Unpack[PrintSettings],
+) -> str:
+    """Produce a Lisp representation of an associative collection,
+    bookended with the start and end string supplied. The entries
+    argument must be a callable which will produce tuples of key-value
+    pairs.
+
+    See basilisp.lang.obj.lepr function for a list of supported
+    keyword arguments.
+
+    The keyword arguments will be passed along to lrepr for the
+    sequence elements.
+
+    """
+
+    def check_same_ns() -> Optional[str]:
+        """Check whether all keys in entries belong to the same
+        namespace. If they do, return the namespace name; otherwise,
+        return None.
+
+        """
+        nses = set()
+        for k, _ in entries():
+            if isinstance(k, INamed):
+                nses.add(k.ns)
+                if len(nses) > 1:
+                    break
+            else:
+                nses.add(None)
+        return next(iter(nses)) if len(nses) == 1 else None
+
+    ns_name_shared = check_same_ns() if kwargs["print_namespace_maps"] else None
+
+    entries_updated = entries
+    if ns_name_shared:
+
+        class EntriesNsRemoved:
+            def __call__(self):
+                for k, v in entries():
+                    yield (k.with_name(k.name), v)
+
+        entries_updated = EntriesNsRemoved()
```

Comment:
Could this class just be a closure?

Reply:
Converted into a fn as suggested, thanks